src/document/json/rga_tree_list.ts

```diff
@@ -174,7 +174,9 @@ export class RGATreeList {
 
     while (
       node!.getNext() &&
-      node!.getNext()!.getCreatedAt().after(executedAt)
+      (node!.getNext()!.getCreatedAt().after(executedAt) ||
+        (node!.getNext()!.getValue().getMovedAt() &&
+          node!.getNext()!.getValue().getMovedAt()!.after(executedAt)))
```

Comment:
It would be nice to test the scenario below.

```
// 1. Push two elements at the same position in the concurrent situation.

initial state: [0]
client 1: [0] -> [1]
client 2: [0] -> [2]
result: [0] -> [1] -> [2] or [0] -> [2] -> [1]

// 2. Move the two elements at the same position in the concurrent situation.

initial state: [0] -> [1] -> [2] (We assume that the result of step 1 is the first case)
client 1: [1] -> [0] -> [2]
client 2: [2] -> [0] -> [1]
result: [1] -> [2] -> [0] or [2] -> [1] -> [0]
```